### PR TITLE
Removed default values for runAsUser, runAsGroup, and fsGroup parameters

### LINKED ISSUE #203: Fix or document `forbidden: unable to validate against any security context constraint`
In OpenShift (and IBM Cloud) environments, the projects/namespaces may be limited to the certain range of user ids. Then,l installing the helm chart results in the following error.
```
forbidden: unable to validate against any security context constraint: [fsGroup: Invalid value: []int64{65534}: 65534 is not an allowed group spec.containers[0].securityContext.securityContext.runAsUser: Invalid value: 65534: must be in the ranges: [1000560000, 1000569999]]
```

We should do the following:
1. Check what other Helm Charts do in such scenarios
2. Find a solution to fix the Helm Chart
3. If not possible, then at least document what to do in such scenarios

### PR DESCRIPTION
**Pull request by alparslanavci**
Without default values, the lowest possible ids are assigned by the container platforms. Thus, there is no need for setting an arbitrary default value.

Fixes #203.

### CODE REVIEW

**Review by leszko — COMMENTED** [review]
Thanks for the PR @alparslanavci 

One question, in `templates/statefulset.yaml`. Will this condition be resolved correctly? Meaning, do we run as `nonRoot` by default now?

```
{{ if eq (int .Values.securityContext.runAsUser) 0 }}false{{ else }}true{{ end }}
```

**Comment by alparslanavci** [comment]
@leszko you are right, it translates none assigned values as `0`. I added `securityContext.runAsNonRoot` parameter and set its default value as `true`.

File: stable/hazelcast-enterprise/templates/mancenter-statefulset.yaml [hunk 1]
```diff
@@ -47,7 +47,7 @@ spec:
       hostIPC: false
       {{- if .Values.securityContext.enabled }}
       securityContext:
```
Comment (leszko) [inline stable/hazelcast-enterprise/templates/mancenter-statefulset.yaml:49]: I see that now we just rewrite exactly every value from `securityContect`, so maybe we should inline it whole like we did with `resources` [here](https://github.com/hazelcast/charts/blob/master/stable/hazelcast/templates/statefulset.yaml#L71)?

I think the code and the documentation will be simpler.

**Review by leszko — COMMENTED** [review]
Added one comment, other than that, it looks fine. 

One more thing, could you check if it works fine on some older k8 cluster (I mean if the users are automatically assigned as non-root)? Older I mean, enough if create the oldest possible on GKE (1.15 I guess). Thanks!

**Pull request closed without merging**